hub.go

```diff
@@ -367,6 +367,15 @@ func GetHubFromContext(ctx context.Context) *Hub {
 	return nil
 }
 
+// hubFromContext returns either a hub stored in the context or the current hub.
+// The return value is guaranteed to be non-nil, unlike GetHubFromContext.
+func hubFromContext(ctx context.Context) *Hub {
```

Comment:
This variation of `GetHubFromContext` is useful in various span methods, because when doing manual instrumentation it makes sense that `StartSpan`, etc, should always find a hub.

I decided not to export for now as users can still use `GetHubFromContext` which has more appropriate semantics in HTTP handlers: handlers should never accidentally use the global hub.